packages/flutter_tools/lib/src/xcode_project.dart

```diff
@@ -357,6 +369,33 @@ class IosProject extends XcodeBasedProject {
         }
       }
     }
+
+    // The build settings of the watchOS companion app's scheme
+    // may contain the key INFOPLIST_KEY_WKCompanionAppBundleIdentifier
+    final bool watchIdentifierFound = firstMatchInFile(xcodeProjectInfoFile, _watchAppCompanionAppIdPattern) != null;
```

Comment:
I don't think this needs to be a regex, you're just checking if the file contains this string at all.  You can also remove `_watchAppCompanionAppIdPattern`.
```suggestion
    final bool watchIdentifierFound = xcodeProjectInfoFile.readAsStringSync().contains('WKCompanionAppBundleIdentifier');
```